### PR TITLE
Delete `gtGetStructHandle` and friends

### PR DESCRIPTION
**Pull request by SingleAccretion**
All of this infrastructure has been made obsolete by the recent changes to support block-typed and SIMD locals without handles.

Includes the deletion of `IsSimdAsHWIntrinsic` logic.

[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=225329&view=ms.vss-build-web.run-extensions-tab) - see below for the explanation.

### CODE REVIEW

File: src/coreclr/jit/fginline.cpp [hunk 1]
```diff
@@ -1654,14 +1654,12 @@ Statement* Compiler::fgInlinePrependStatements(InlineInfo* inlineInfo)
 
                     if (varTypeIsStruct(argType))
                     {
-                        structHnd = gtGetStructHandleIfPresent(argNode);
-                        noway_assert((structHnd != NO_CLASS_HANDLE) || (argType != TYP_STRUCT));
+                        structHnd = lclVarInfo[argNum].lclVerTypeInfo.GetClassHandleForValueClass();
+                        assert(structHnd != NO_CLASS_HANDLE);
```
Comment (SingleAccretion) [inline src/coreclr/jit/fginline.cpp:1658]: This is the reason for the few diffs: previously, the struct handle could be "refined" to a type more specific than `f(__Canon)`, which affects VN-based copy propagation.

The diffs could be fixed (i. e. made to not exist), but only with quirks, which I did not find worth it due to the already-not-small size of this change.

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib

File: src/coreclr/jit/compiler.h [hunk 2]
```diff
@@ -8418,253 +8385,21 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
...
-
-        // SIMD Types
-        CORINFO_CLASS_HANDLE VectorTHandles[SupportedTypeCount];
-
         CORINFO_CLASS_HANDLE PlaneHandle;
         CORINFO_CLASS_HANDLE QuaternionHandle;
         CORINFO_CLASS_HANDLE Vector2Handle;
         CORINFO_CLASS_HANDLE Vector3Handle;
         CORINFO_CLASS_HANDLE Vector4Handle;
         CORINFO_CLASS_HANDLE VectorHandle;
```
Comment (tannergooding) [inline src/coreclr/jit/compiler.h:8393]: Why do these ones need to stay?

Comment (tannergooding) [reply src/coreclr/jit/compiler.h:8393]: Is it just for `isOpaqueSIMDType`? Is there a plan to also remove that somehow?

**Review by tannergooding — APPROVED** [review]

File: src/coreclr/jit/compiler.h (hunk 2 above)
Comment (SingleAccretion) [reply src/coreclr/jit/compiler.h:8393]: > Is it just for `isOpaqueSIMDType`?

Yep. No explicit plans for its removal currently, though one could certainly imagine rewriting it to not need the cached handles (e. g. a new `getClassAttribs` flag, or matching by name).

**Comment by AndyAyersMS** [comment]
@SingleAccretion can you fix up the merge conflicts?

**Comment by SingleAccretion** [comment]
Conflicts have been fixed.

**Comment by AndyAyersMS** [comment]
Linux arm64 failure looks like an instance of #82771 or similar, hard to be sure without artifacts.

I thkink the SPMI failures are also known issues -- @kunalspathak FYI
```
[20:35:14] ISSUE: <ASSERT> #463912 D:\a\_work\1\s\src\coreclr\jit\lsra.cpp (11880) - Assertion failed '!"Spill candidate has no assignedInterval recentRefPosition"' in 'JIT.HardwareIntrinsics.Arm._AdvSimd.VectorLookup_3Test__VectorTableLookupSByte:RunBasicScenario_UnsafeRead():this' during 'LSRA build intervals' (IL size 169; hash 0xe0ad7041; FullOpts)
```

**Comment by tannergooding** [comment]
https://github.com/dotnet/runtime/issues/84536 Is the Arm64 SPMI failure

**Comment by AndyAyersMS** [comment]
@SingleAccretion thank you.

**Pull request merged**